modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -452,13 +452,23 @@ public Hit getHitInfo(float x, float y) {
                 charIndex = run.getStart() + run.getOffsetAtX(x, trailing);
                 leading = (trailing[0] == 0);
 
-                // Calculate insertion index for surrogate pair text in TextFlow
-                // when leading is false
-                if (run.getTextSpan() != null) {
-                    String txt = run.getTextSpan().getText();
-                    int relIdx = charIndex - run.getStart();
-                    if (Character.isHighSurrogate(txt.charAt(relIdx)) && !leading) {
-                        insertionIndex = charIndex + 2;
+                insertionIndex = charIndex;
+                String txt = new String(getText());
+                if (!leading) {
+                    if (txt != null) {
+                        int next;
+                        BreakIterator charIterator = BreakIterator.getCharacterInstance();
+                        synchronized(charIterator) {
```

Comment:
question: why do we need to synchronize on the instance here?  BreakIterator.get*Instance() creates a new instance each time, so synchronization is probably unnecessary.

Reply:
Correct it is not necessary. Updated code to remove synchronize